WordPress/WordPressTest/PostEditorStateTests.swift

```diff
@@ -0,0 +1,149 @@
+import XCTest
+@testable import WordPress
+
+class PostEditorStateTests: XCTestCase {
+    var context: PostEditorStateContext!
+
+    override func setUp() {
+        super.setUp()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        context = nil
+    }
+
+    func testContextNoContentPublishButtonDisabled() {
+        context = PostEditorStateContext(originalPostStatus: .publish, userCanPublish: true, delegate: self)
+
+        context.updated(hasContent: false)
+
+        XCTAssertFalse(context.isPublishButtonEnabled)
+    }
+
+    func testContextChangedNewPostToDraft() {
+        context = PostEditorStateContext(originalPostStatus: .draft, userCanPublish: true, delegate: self)
```

Comment:
I fixed this with a check-in - new posts will pass in a nil originalPostStatus now since it's not on the server. I was trying to match logic in Calypso which makes every post a draft when creating it which is something we should eventually emulate. 😄